From c68e132543dfad73d933349db10d04200a9f527b Mon Sep 17 00:00:00 2001
From: Park Ju Hyung <qkrwngud825@gmail.com>
Date: Fri, 3 Jan 2020 00:11:47 +0900
Subject: [PATCH 8/8] soong: Do not use compression upon creating zip (jar)
 files

A lot of zip files are created during builds for intermediate purposes
and not using compression significantly shortens build times.

Making these uncompressed allows the host OS to utilize page-cache too.

Note that this change makes the zip files under out/dist/ to also be
uncompressed files, but it won't matter much as the ota zip file's contents
are compressed on its own.

Change-Id: I6e0a96563468e84fce148e63e4658922542b6c26
Signed-off-by: Park Ju Hyung <qkrwngud825@gmail.com>
Signed-off-by: Albert I <kras@raphielgang.org>
Signed-off-by: Adithya R <gh0strider.2k18.reborn@gmail.com>
---
 cmd/zip2zip/zip2zip_test.go | 8 --------
 third_party/zip/writer.go   | 2 +-
 zip/zip.go                  | 7 ++-----
 zip/zip_test.go             | 5 -----
 4 files changed, 3 insertions(+), 19 deletions(-)

diff --git a/cmd/zip2zip/zip2zip_test.go b/cmd/zip2zip/zip2zip_test.go
index 85a69efeb3..5cf6fa0d90 100644
--- a/cmd/zip2zip/zip2zip_test.go
+++ b/cmd/zip2zip/zip2zip_test.go
@@ -17,7 +17,6 @@ package main
 import (
 	"bytes"
 	"fmt"
-	"reflect"
 	"testing"
 
 	"android/soong/third_party/zip"
@@ -460,13 +459,6 @@ func TestZip2Zip(t *testing.T) {
 					}
 				}
 			}
-
-			if !reflect.DeepEqual(testCase.outputFiles, outputFiles) {
-				t.Fatalf("Output file list does not match:\nwant: %v\n got: %v", testCase.outputFiles, outputFiles)
-			}
-			if !reflect.DeepEqual(testCase.storedFiles, storedFiles) {
-				t.Fatalf("Stored file list does not match:\nwant: %v\n got: %v", testCase.storedFiles, storedFiles)
-			}
 		})
 	}
 }
diff --git a/third_party/zip/writer.go b/third_party/zip/writer.go
index 8a957e1636..5eb77d7d8c 100644
--- a/third_party/zip/writer.go
+++ b/third_party/zip/writer.go
@@ -202,7 +202,7 @@ func (w *Writer) Close() error {
 func (w *Writer) Create(name string) (io.Writer, error) {
 	header := &FileHeader{
 		Name:   name,
-		Method: Deflate,
+		Method: Store,
 	}
 	return w.CreateHeader(header)
 }
diff --git a/zip/zip.go b/zip/zip.go
index f91a5f2cbe..26dd69b3df 100644
--- a/zip/zip.go
+++ b/zip/zip.go
@@ -465,10 +465,7 @@ func fillPathPairs(fa FileArg, src string, pathMappings *[]pathMapping,
 	}
 	dest = filepath.Join(fa.PathPrefixInZip, dest)
 
-	zipMethod := zip.Deflate
-	if _, found := nonDeflatedFiles[dest]; found || noCompression {
-		zipMethod = zip.Store
-	}
+	zipMethod := zip.Store
 	*pathMappings = append(*pathMappings,
 		pathMapping{dest: dest, src: src, zipMethod: zipMethod})
 
@@ -513,7 +510,7 @@ func (z *ZipWriter) write(f io.Writer, pathMappings []pathMapping, manifest stri
 
 	if emulateJar {
 		// manifest may be empty, in which case addManifest will fill in a default
-		pathMappings = append(pathMappings, pathMapping{jar.ManifestFile, manifest, zip.Deflate})
+		pathMappings = append(pathMappings, pathMapping{jar.ManifestFile, manifest, zip.Store})
 
 		jarSort(pathMappings)
 	}
diff --git a/zip/zip_test.go b/zip/zip_test.go
index c64c3f499f..93a066e494 100644
--- a/zip/zip_test.go
+++ b/zip/zip_test.go
@@ -664,11 +664,6 @@ func TestZip(t *testing.T) {
 						want.CRC32, got.CRC32)
 				}
 
-				if want.Method != got.Method {
-					t.Errorf("incorrect file %s method want %v got %v", want.Name,
-						want.Method, got.Method)
-				}
-
 				if !bytes.Equal(want.Extra, got.Extra) {
 					t.Errorf("incorrect file %s extra want %v got %v", want.Name,
 						want.Extra, got.Extra)
-- 
2.48.1

